### PR TITLE
Correct bitcode support for iOS

### PR DESCRIPTION
**Pull request by ikliashchou**
Don't know how to squash everything in one commit :( there was a conflict and some other changes in between

### CODE REVIEW

**Comment by ikliashchou** [comment]
It seems that Travis tests failed because of cmake update:

`File not exists: /Users/travis/build/ruslo/polly/_ci/cmake-3.9.0-rc6-Darwin-x86_64.tar.gz`

**Comment by ruslo** [comment]
> Don't know how to squash everything in one commit :(

The simplest way to remember is to do `git reset` and add/commit everything from scratch.
See: https://stackoverflow.com/a/5201642/2288008

> It seems that Travis tests failed because of cmake update

No it's not the reason, reason is
```
No output has been received in the last 10m0s, this potentially indicates
a stalled build or something wrong with the build itself.
```

I've restarted this job.

**Comment by ikliashchou** [comment]
Used this technic :) 
https://stackoverflow.com/questions/9646167/clean-up-a-fork-and-restart-it-from-the-upstream

File: flags/bitcode.cmake [hunk 1]
```diff
@@ -7,12 +7,7 @@ else()
...
+set(CMAKE_XCODE_ATTRIBUTE_BITCODE_GENERATION_MODE "bitcode")
 
-polly_add_cache_flag(CMAKE_CXX_FLAGS "-fembed-bitcode")
-
-# We can't use 'ENABLE_BITCODE' because
-# it only adds '-fembed-bitcode-marker':
-# * https://stackoverflow.com/a/31346742
-
-list(APPEND HUNTER_TOOLCHAIN_UNDETECTABLE_ID "bitcode.2")
+list(APPEND HUNTER_TOOLCHAIN_UNDETECTABLE_ID "bitcode.2")
```
Comment (ruslo) [inline flags/bitcode.cmake]: "No newline at end of file" warning:
* http://0.readthedocs.io/en/latest/git.html#check-there-is-a-new-line-at-end-of-file

Comment (ruslo) [inline flags/bitcode.cmake]: It should be `bitcode.3` since implementation changed.

**Pull request closed without merging**